crates/shared/src/sources/balancer_v2/graph_api.rs

```diff
@@ -485,8 +486,10 @@ mod tests {
     async fn balancer_subgraph_query() {
         for (network_name, chain_id) in [("Mainnet", 1), ("Goerli", 5)] {
             println!("### {network_name}");
-
-            let client = BalancerSubgraphClient::for_chain(chain_id, Client::new()).unwrap();
+            let base_url =
+                Url::parse("https://api.thegraph.com/subgraphs/name/").expect("invalid url");
```

Comment:
nit: If you don't want manually change all the test files you could expose an additional `::default_for_chain` method on the SubgraphClient that uses the default URL and is used in the tests.